File: Monitor-Shop-FrontEnd/src/app/common/Rate.ts
import { Customer } from "./Customer";
import { Product } from "./Product";

export class Rate {
    'id': number;
    'star': number;
    'comment': string;
    'rateDate': Date;
    'product': Product;
    'user': Customer;

    constructor(id:number, star:number, comment:string, rateDate: Date, product:Product, user: Customer) {
        this.id = id;
        this.star = star;
        this.comment = comment;
        this.rateDate = rateDate;
        this.product = product;
        this.user = user;
    }
}

File: Monitor-Shop-BackEnd/src/main/java/com/mshop/entity/Rate.java
package com.mshop.entity;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "rates")
public class Rate {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private Double star;
	private String comment;
	private Date rateDate;
	
	@ManyToOne
	@JoinColumn(name = "userId")
	private User user;
	
	@ManyToOne
	@JoinColumn(name = "productId")
	private Product product;
}
